util/file.go

```diff
@@ -253,20 +256,68 @@ func JoinTerraformModulePath(modulesFolder string, path string) string {
 	return fmt.Sprintf("%s//%s", cleanModulesFolder, cleanPath)
 }
 
-// Returns true if the given path contains the given folder name.
-//
-// Examples:
-//
-// PathContains("/foo/bar", "foo") => returns true
-// PathContains("/foo/bar", "baz") => returns false
-func PathContains(path string, folderName string) bool {
-	pathParts := strings.Split(path, string(filepath.Separator))
+type fileManifest struct {
+	Path       string
+	encoder    *gob.Encoder
+	fileHandle *os.File
+}
 
-	for _, pathPart := range pathParts {
-		if pathPart == folderName {
-			return true
+// Clean will remove all files specified in the manifest
+func (f *fileManifest) Clean() error {
+	var path string
+
+	// if manifest file doesn't exist, just exit
+	if !FileExists(f.Path) {
+		return nil
+	}
+	file, err := os.Open(f.Path)
+	if err != nil {
+		return err
+	}
+	decoder := gob.NewDecoder(file)
```

Comment:
Sorry, I'm probably missing something obvious, but what is the role of the `decoder` here?

Reply:
The whole point of `gob` is that it provides ability to decode filenames one by one, instead of load the file in memory at once.